File: Backend/WebAPI/Src/Core/Entities/Learner.cs
using Core.Enums;

namespace Core.Entities;

public class Learner : BaseUser
{
    public required RankType Rank { get; set; }
    public uint Point { get; set; }
    public uint BestScore { get; set; }
    public uint WorstScore { get; set; }
}

File: Backend/WebAPI/src/main/java/com/vhl/webapi/entities/specific/Learner.java
package com.vhl.webapi.entities.specific;

import com.vhl.webapi.enums.LearnerRank;
import com.vhl.webapi.enums.Role;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "learners")
@PrimaryKeyJoinColumn(name = "id")
public class Learner extends BaseUser {
    @Enumerated(EnumType.STRING)
    @Column(name = "learner_rank", nullable = false)
    private LearnerRank rank = LearnerRank.BEGINNER;

    @Column(name = "point", nullable = false)
    private Integer point = 0;

    @Column(name = "best_score")
    private Double bestScore;

    @Column(name = "worst_score")
    private Double worstScore;


    @Override
    public String getFullRole() {
        return Role.LEARNER.name();
    }
}
